db/account.go

```diff
@@ -264,3 +264,102 @@ func FetchUsernameFromUUID(uuid []byte) (string, error) {
 
 	return username, nil
 }
+
+func FetchUUIDFromUsername(username string) ([]byte, error) {
+	var uuid []byte
+	err := handle.QueryRow("SELECT uuid FROM accounts WHERE username = ?", username).Scan(&uuid)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			// No rows is not an error in this context, it signifies that the user does not exists
+			// we reserve err for db errors.
+			return uuid, nil
+		}
+		return nil, err
+	}
+
+	return uuid, nil
+}
+
+func IsFriendWith(sourceUuid []byte, friendUuid []byte) (bool, error) {
+	var result int
+	err := handle.QueryRow("SELECT COUNT(*) FROM friends WHERE user = ? AND friend = ?", sourceUuid, friendUuid).Scan(&result)
+	if err != nil {
+		return false, err
+	}
+
+	return result == 1, nil
+}
+
+func AddFriend(uuid []byte, friendUsername string) (bool, error) {
+	// We are making db errors more generic as error is used in the response data to the client.
+	friendUuid, err := FetchUUIDFromUsername(friendUsername)
+	if err != nil {
+		return false, fmt.Errorf("An error occured, if this persist, please contact an administrator.")
+	}
+
+	if len(friendUuid) == 0 {
+		return false, fmt.Errorf("User does not exist")
+	}
+
+	alreadyFriends, _ := IsFriendWith(uuid, friendUuid)
```

Comment:
just create an unique constraint on db so only one `(user, friend)` pair is allowed